src/helpers/generator.ts

```diff
@@ -8,36 +8,126 @@ import { convertKeyValueToLine } from "../writer/stringBuilder";
 
 // TODO next: Pull this file apart. Create a new "file writer" file, move all the ts stuff somewhere else,
 
-// this strips comments of special tokens since ts-morph generates jsdoc tokens automatically
-const cleanComment = (comment: string) => {
+export const cleanComment = (comment: string): string => {
+  if (!comment) return "";
+  if (comment.trim() === "/** */") return "";
+
   return comment
-    .replace(/^\/\*\*[^\S\r\n]?/, "")
-    .replace(/[^\S\r\n]+\*\s/g, "")
-    .replace(/(\n)?[^\S\r\n]+\*\/$/, "");
+    .replace(/^\/\*\*[^\S\r\n]?/, "") // Remove opening /**
+    .replace(/[^\S\r\n]+\*\s/g, "") // Remove * at start of lines
+    .replace(/(\n)?[^\S\r\n]+\*\/$/, ""); // Remove closing */
 };
 
-const convertFuncSignatureToType = (
+export const sanitizeTypeName = (name: string): string => {
+  try {
+    // First, handle empty or invalid input
+    if (!name) {
+      console.warn('sanitizeTypeName received empty name, using default "UnknownType"');
+      return "UnknownType";
+    }
+
+    if (typeof name !== "string") {
+      console.warn(
+        `sanitizeTypeName received non-string input: ${typeof name}, using default "UnknownType"`
+      );
+      return "UnknownType";
+    }
+
+    const parts = name
```

Comment:
Can you explain the rational for splitting the name into parts and validating each part? I dont think it's our job to dictate what a users collection name should be, as long as we can generate a model name for it I think that's all that matters. I would simply  upper case the first character of the string and after any period, remove any periods and leave the rest. Is there anything else Im missing there that we need?

Reply:
I guess it depends on the direction you want to take the tool. We can go pretty deep on sanitization here but there is definitely an argument against adding too much "magic" auto-parsing. 

for example we could allow reserved words instead of throwing by prefacing each one with a 'I' like:

```typescript
// not valid
type super =...
// valid 
type Isuper =...
```
In my experience it's hard to maintain documentation on things like this.

That being said, the alternative is broken output files and when the user cannot easily change collection name (like me) it makes more sense to add the magic in.

lmk what you think (also no rationale on the name splitting, just sloppy code :) )

Reply:
Yeah I see the case where you may not be able to change the model name yourself. 

Okay ill add my thoughts on the new code you added